servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1ClientDuplexHandler.java

```diff
@@ -166,17 +166,16 @@ private NettyH2HeadersToHttpHeaders h2HeadersToH1HeadersClient(Http2Headers h2He
         h2HeadersSanitizeForH1(h2Headers);
         if (httpStatus != null) {
             final int statusCode = httpStatus.code();
-            final Long contentLength = h2Headers.getLong(CONTENT_LENGTH);
-            if (contentLength == null) {
-                if (fullResponse) {
-                    if (responseMayHaveContent(statusCode, method)) {
+            if (!h2Headers.contains(CONTENT_LENGTH)) {
+                if (serverMaySendPayloadBodyFor(statusCode, method)) {
+                    if (fullResponse) {
                         h2Headers.set(CONTENT_LENGTH, ZERO);
+                    } else {
+                        h2Headers.add(TRANSFER_ENCODING, CHUNKED);
                     }
-                } else if (canAddResponseTransferEncodingProtocol(statusCode, method)) {
-                    h2Headers.add(TRANSFER_ENCODING, CHUNKED);
                 }
             } else if (!responseMayHaveContent(statusCode, method)) {
-                throw new IllegalArgumentException("content-length (" + contentLength +
+                throw new IllegalArgumentException("content-length (" + h2Headers.getLong(CONTENT_LENGTH) +
```

Comment:
just do `h2Headers.get(CONTENT_LENGTH)` bcz we are using it as a string here anyways.